README.md

```diff
@@ -62,18 +78,67 @@ public class Compute {
+
+The following code-snippet shows the naive Matrix Multiplication using the kernel-parallel API:
+
+```java
+public class Compute {
+    private static void mxmKernel(TornadoVMContext context, Matrix2DFloat A, Matrix2DFloat B, Matrix2DFloat C, final int size) {
+        int idx = context.threadIdx;
+        int jdx = context.threadIdy;
+        float sum = 0;
+        for (int k = 0; k < size; k++) {
+            sum += A.get(idx, k) * B.get(k, jdx);
+        }
+        C.set(idx, jdx, sum);
+    }
+
+    public void run(Matrix2DFloat A, Matrix2DFloat B, Matrix2DFloat C, final int size) {
+        // When using the kernel-parallel API, we need to crete a Grid and a Worker
+
+        WorkerGrid workerGrid = new WorkerGrid2D(size, size);    // Create 2D Worker
+        GridTask gridTask = new GridTask("s0.t0", workerGrid);   // Attach the worker to the Grid
+        TornadoVMContext context = new TornadoVMContext();       // Create a context
+        workerGrid.setLocalWork(32, 32, 1);                      // Set the local-group size
+
+        TaskSchedule ts = new TaskSchedule("s0")
+                .streamIn(A, B)                                 // Stream data from host to device
+                .task("t0", Compute::mxmKernel, context, A, B, C, size)  // Each task points to an existing Java method
+                .streamOut(C);                                  // sync arrays with the host side
+        ts.execute(gridTask);   // Execute with a GridTask
+    }
+}
+```
+
+Additionally, the two modes of expressing parallelism (kernel and loop parallelization) can be combined in the same
+task-schedule object.
 
 ## 4. Dynamic Reconfiguration
 
-Dynamic reconfiguration is the ability of TornadoVM to perform live task migration between devices, which means that TornadoVM decides where to execute the code to increase performance (if possible). In other words, TornadoVM switches devices if it knows the new device offers better performance. With the task-migration, the TornadoVM's approach is to only switch device if it detects an application can be executed faster than the CPU execution using the code compiled by C2 or Graal-JIT, otherwise it will stay on the CPU. So TornadoVM can be seen as a complement to C2 and Graal. This is because there is no single hardware to best execute all workloads efficiently. GPUs are very good at exploiting SIMD applications, and FPGAs are very good at exploiting pipeline applications. If your applications follow those models, TornadoVM will likely select heterogeneous hardware. Otherwise, it will stay on the CPU using the default compilers (C2 or Graal).
+Dynamic reconfiguration is the ability of TornadoVM to perform live task migration between devices, which means that
+TornadoVM decides where to execute the code to increase performance (if possible). In other words, TornadoVM switches
+devices if it knows the new device offers better performance. With the task-migration, the TornadoVM's approach is to
```

Comment:
if it can detect that a specific device can yield better performance (compared to another)